core/server/common/src/main/java/alluxio/cli/validation/UserLimitValidationTask.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.validation;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Task for validating system limit for current user.
+ */
+public final class UserLimitValidationTask implements ValidationTask {
+  private static final int NUMBER_OF_OPEN_FILES_MIN = 4096;
```

Comment:
We don't have a strict formula for calculating such numbers in the docs, but it's more like a recommendation. In the [configuration page](http://www.alluxio.org/docs/master/en/Configuration-Properties.html), you can find several entries on the number of threads Alluxio will create. For example, `alluxio.master.worker.threads.max` denotes the number of threads for handling incoming RPC. So we can get a sense of max user processes might be needed for the.
Also, Alluxio is recommended to collocate with the compute framework, which means we also need to estimate some resources for compute framework use. 
That's why some big number of open files and user processes are recommended.